s4e-backend/src/main/java/pl/cyfronet/s4e/service/ReportTemplateMapper.java

```diff
@@ -0,0 +1,27 @@
+package pl.cyfronet.s4e.service;
+
+import lombok.extern.slf4j.Slf4j;
+import org.mapstruct.Mapper;
+import org.mapstruct.Mapping;
+import org.springframework.beans.factory.annotation.Autowired;
+import pl.cyfronet.s4e.bean.AppUser;
+import pl.cyfronet.s4e.bean.ReportTemplate;
+import pl.cyfronet.s4e.config.MapStructCentralConfig;
+import pl.cyfronet.s4e.data.repository.AppUserRepository;
+import pl.cyfronet.s4e.ex.NotFoundException;
+
+@Mapper(config = MapStructCentralConfig.class)
```

Comment:
What does this code? Is it only finding user by email? If yes, why isn't placed in AppUserService?

Reply:
It's the global MapStruct config actually.

The class, on the other hand, as it's name suggests, maps ReportTemplate transfer classes onto themselves. The one abstract method is in fact named incorrectly, it should be `createDtoToPreEntity` as the request is mapped by hand.
The user finding is a mapping operation, which fits nice in the mapper, simplifying the `ReportTemplateService`. This logic shouldn't be placed in `AppUserService` as in general services shouldn't use other services, except some corner cases, but refer directly to the repositories.